ServiceWebsite/ServiceWebsite/Controllers/ParticipantController.cs

```diff
@@ -34,8 +42,75 @@ public ParticipantController()
         [ProducesResponseType((int)HttpStatusCode.BadRequest)]
         public async Task<IActionResult> UpdateSuitabilityAnswers(Guid hearingId, [FromBody]List<HearingSuitabilityAnswer> answers)
         {
-          await Task.Yield();
-          return NoContent();
+            if (hearingId == Guid.Empty)
+            {
+                return new BadRequestObjectResult($"Please provide a valid {nameof(hearingId)}");
+            }
+            try
+            {
+                var participantId = await _hearingService.GetParticipantIdAsync(User.Identity.Name, hearingId);
+
+                if (!participantId.HasValue || participantId == Guid.Empty)
+                {
+                    return new UnauthorizedObjectResult($"User is not a participant of hearing with id '{hearingId}'");
+                }
+                
+                if (answers.Count == 0)
+                {
+                    return new BadRequestObjectResult($"Please provide a valid answers");
+                }
+                else
+                {
+                    foreach (HearingSuitabilityAnswer answer in answers)
+                    {
+                        bool isValid = ValidateAnswerKey(answer.QuestionKey);
+                        if (!isValid)
+                        {
+                            return new BadRequestObjectResult($"Please provide a valid answer key {nameof(answer.QuestionKey)}");
```

Comment:
Might be good if it explains what a `valid answer key` is, it's not important but it's good to explain to the client not just what is wrong but also why

Reply:
Resolved